Buildings/Controls/OBC/ASHRAE/G36_PR1/Generic/SetPoints/ZoneStatus.mo

```diff
@@ -1,380 +1,420 @@
-    annotation (Placement(transformation(extent={{80,150},{100,170}})));
-  Buildings.Controls.OBC.CDL.Continuous.Add add(final k2=-1)
-    "Calculate differential between minimum zone temperature and the heating setpoint"
-    annotation (Placement(transformation(extent={{-20,60},{0,80}})));
+    "Decide if the warm-up time of one zone should be ignored: if window is open, 
+    then output zero, otherwise, output warm-up time from optimal warm-up block"
+    annotation (Placement(transformation(extent={{120,170},{140,190}})));
+  Buildings.Controls.OBC.CDL.Continuous.Add add(
+    final k2=-1)
+    "Calculate difference between zone temperature and the occupied heating setpoint"
+    annotation (Placement(transformation(extent={{-40,80},{-20,100}})));
   Buildings.Controls.OBC.CDL.Continuous.Hysteresis hys(
-    final pre_y_start=false,
     final uLow=-0.1,
     final uHigh=0.1)
     "Hysteresis that outputs if the system should run in warm-up mode"
-    annotation (Placement(transformation(extent={{20,60},{40,80}})));
-  Buildings.Controls.OBC.CDL.Continuous.Add add1(final k1=-1)
-    "Calculate differential between maximum zone temperature and the cooling setpoint"
-    annotation (Placement(transformation(extent={{-20,10},{0,30}})));
+    annotation (Placement(transformation(extent={{0,80},{20,100}})));
+  Buildings.Controls.OBC.CDL.Continuous.Add add1(
+    final k1=-1)
+    "Calculate difference between zone temperature and the occupied cooling setpoint"
+    annotation (Placement(transformation(extent={{-40,0},{-20,20}})));
   Buildings.Controls.OBC.CDL.Continuous.Hysteresis hys1(
-    final pre_y_start=false,
     final uLow=-0.1,
     final uHigh=0.1)
     "Hysteresis that outputs if the system should run in cool-down mode"
-    annotation (Placement(transformation(extent={{20,10},{40,30}})));
-  Buildings.Controls.OBC.CDL.Continuous.Add add2(final k1=-1)
-    "Calculate zone temperature difference to setpoint"
-    annotation (Placement(transformation(extent={{40,-70},{60,-50}})));
+    annotation (Placement(transformation(extent={{0,0},{20,20}})));
+  Buildings.Controls.OBC.CDL.Continuous.Add add2(
+    final k2=-1)
+    "Calculate zone temperature difference to unoccupied heating setpoint"
+    annotation (Placement(transformation(extent={{-40,-80},{-20,-60}})));
   Buildings.Controls.OBC.CDL.Continuous.Hysteresis hys2(
-    final pre_y_start=false,
     final uLow=-0.1,
     final uHigh=0.1)
-    "Hysteresis that outputs if the zone temperature is lower then setpoint"
-    annotation (Placement(transformation(extent={{80,-70},{100,-50}})));
-  Buildings.Controls.OBC.CDL.Continuous.Add add3(final k2=-1)
-    "Calculate the difference between minimum zone temperature and unoccupied heating setpoint"
-    annotation (Placement(transformation(extent={{40,-120},{60,-100}})));
-  Buildings.Controls.OBC.CDL.Continuous.Hysteresis hys3(
-    final uLow=-0.5*bouLim,
-    final uHigh=0.5*bouLim,
```

Comment:
uLow of hys2 and hys3 should remain the same so that the setback mode can be activated. So does uHigh of hys2 and hys3. For this one, you can set one parameter bouLim or set two parameters separately uLow and uHigh as I suggested for hys and hys1. This is up to you to decide.

Reply:
@kuzha 
The `hy3` and `hy4` are different from other hysteresis. Here we'd need to check if the temperature is higher/lower than setpoints by 1 degC. 
Other hysteresis check if temperatures is higher/lower than setpoints.